### DISCUSSION TITLE
Separate FQDN for server and application?

### DISCUSSION THREAD

**Discussion by JoWe112** [uyuni-project/uyuni#9136] [Q&A] [answered]
Hi.
When I deployed Uyuni 2024.07 on openSUSE Leap the FQDN for Unify will be the server FQDN. Isn't it possible to specify a different FQDN for the "application" rather then use the server FQDN?

/ Joakim

**Comment by rjmateus** [comment, marked as answer]
are you using the containerized version of uyuni? If yes, then you can add a parameter at the end of the install to set the FQDN you want to be configured. It will influence the generated certificates.
Example: `mgradm install podman my_special_fqdn`

**Reply by sbluhm** [reply]
Thank you @rjmateus , is there a way to also change the fqdn that way? I couldn't find the option.

**Reply by rjmateus** [reply]
I'm not sure if I got your question, your goal is to change the FQDN after installation?

**Reply by sbluhm** [reply]
Yes, correct. I installed Uyuni without the FQDN paramter (now it is podman.example.com). Now I would like to change it to uyuni.example.com.

**Reply by rjmateus** [reply]
the old tools are still available inside the container. You should be able to follow it: https://www.uyuni-project.org/uyuni-docs/en/uyuni/administration/troubleshooting/tshoot-hostname-rename.html

I'm not sure it this procedure was tested in the containerized server.

**Reply by sbluhm** [reply]
I think this won't work based on the usage:
```
# spacewalk-hostname-rename
Missing <ip_address> argument.
Usage:
   spacewalk-hostname-rename <IP_ADDRESS> [ --ssl-country=<SSL_COUNTRY> --ssl-state=<SSL_STATE> --ssl-city=<SSL_CITY> --ssl-org=<SSL_ORG> --ssl-orgunit=<SSL_ORGUNIT> --ssl-email=<SSL_EMAIL> --ssl-ca-password=<SSL_CA_PASSWORD> --ssl-ca-cert=<PATH> --ssl-server-key=<PATH> --ssl-server-cert=<PATH> --overwrite_report_db_host=[y/n]]
```
From my understanding, `spacewalk-hostname-rename` will be doing a reverse lookup of the IP (which will then find the container server's host name). An individual host name does not seem to be there.
Previously, I tried running the commands inside of  `spacewalk-hostname-rename`  manually and got it somewhat renamed. The push of the susemanager repo configuration has the old server name and I am not sure if cobbler worked correctly.
But I think, this might be a different topic/thread.

General renewing of certificates won't be an issue as there is no reverse lookup, correct?

**Comment by JoWe112** [comment]
I found a bug during my testing...

Since I installed and uninstalled several times I noticed that `mgradm install podman` don't overwrite existing `/etc/systemd/system/uyuni-servic*` files and directorys so the container is not created.

The error i see is `Enabling system service Error: cannot wait for system start: failed to generate systemd service unit file: /etc/systemd/system/uyuni-server.service file already present, not overwriting`

The solution/workaround is to run `mgradm uninstall --force --prune-volumes`and then `rm -rf /etc/systemd/system/uyuni-server.servic*`

Now can I run `mgradm install podman`

Worth to know is that i use openLEAP Micro 6.0

**Reply by rjmateus** [reply]
Are you using the latest version for mgradm? We had a issue like that in the past but it was solved. The uninstall should remove all the systemd information/fileS.

**Reply by JoWe112** [reply]
```
mgradm -version
mgradm version 0.1.17 (Master 311105e)
```
